openspp-project/openspp-registry
# Restructure the spp_user_roles module to support dynamic setup of user roles by other modules

## **Why is this change needed?**
The OpenSPP User Roles module is currently set to depend on other modules for the setup of user roles. An issue is introduced by this considering that there is no flexibility in managing variations in access rights based on the implementation requirements. For example, the SP-MIS registrar role have a different access rights defined compared with the registrar role in the Farmer Registry. Since all modules in the spp_user_roles defined in its dependencies will be installed automatically, there is another issue of having modules not applicable for an implementation to be installed. 

## **How was the change implemented?**

**1. Remove all module specific role definitions in spp_user_roles and move it to the main/base modules:**
- [x] SPP Base module for the SPMIS registry
- [x] SPP Programs module for the Program management

**2. Remove non-generic module dependencies in the spp_user_roles module.**

## **New unit tests**
None

## **Unit tests executed by the author**
None

## **How to test manually**

- Login as the super user.
- Create a new user then select the role to be tested.
- Assign a password for the user.
- Depending on the role assigned to the new user, they should be able to log in and access all the components associated with the role.

## **Related links**

- [Restructure the spp_user_roles module to support dynamic setup of user roles by other modules](https://github.com/OpenSPP/openspp-modules/issues/585)
